File: serialization/typescript/json/src/jsonSerializationWriterFactory.ts
import { SerializationWriter, SerializationWriterFactory } from "@microsoft/kiota-abstractions";
import { JsonSerializationWriter } from "./jsonSerializationWriter";

export class JsonSerializationWriterFactory implements SerializationWriterFactory {
    public getValidContentType() : string { return "application/json"; }
    public getSerializationWriter(contentType: string): SerializationWriter {
        if(!contentType) {
            throw new Error("content type cannot be undefined or empty");
        } else if (this.getValidContentType() !== contentType) {
            throw new Error(`expected a ${this.getValidContentType()} content type`);
        }
        return new JsonSerializationWriter();
    }
}

File: serialization/go/json/json_serialization_writer_factory.go
package jsonserialization

import (
	"errors"

	absser "github.com/microsoft/kiota/abstractions/go/serialization"
)

// The SerializationWriterFactory implementation for JSON serialization.
type JsonSerializationWriterFactory struct {
}

// Creates a new instance of the JsonSerializationWriterFactory.
func NewJsonSerializationWriterFactory() *JsonSerializationWriterFactory {
	return &JsonSerializationWriterFactory{}
}

func (f *JsonSerializationWriterFactory) GetValidContentType() (string, error) {
	return "application/json", nil
}
func (f *JsonSerializationWriterFactory) GetSerializationWriter(contentType string) (absser.SerializationWriter, error) {
	validType, err := f.GetValidContentType()
	if err != nil {
		return nil, err
	} else if contentType == "" {
		return nil, errors.New("contentType is empty")
	} else if contentType != validType {
		return nil, errors.New("contentType is not valid")
	} else {
		return NewJsonSerializationWriter(), nil
	}
}
